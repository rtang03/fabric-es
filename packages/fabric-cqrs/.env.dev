CHANNEL_NAME=loanapp
CA_NAME=rca-org1
CA_ENROLLMENT_ID_ADMIN=rca-org1-admin
CA_ENROLLMENT_SECRET_ADMIN=rca-org1-adminPW
ORDERER_HOSTNAME=orderer0-org0
ORDERER_TLSCA_CERT=../../deployments/dev-net/artifacts/crypto-config/Org0MSP/orderer0.org0.com/tls-msp/tlscacerts/tls-0-0-0-0-5052.pem
CONNECTION_PROFILE=connection/connection-org1.local.yaml
MSPID=Org1MSP
ORG_ADMIN_ID=admin-org1.net
ORG_ADMIN_SECRET=admin-org1.netPW
WALLET=assets/wallet

# Org2 (pbctfp) - SOLELY used for running service.multiusers.test.ts
ORG2_ORG_ADMIN_ID=admin-org2.net
ORG2_ORG_ADMIN_SECRET=admin-org2.netPW
ORG2_MSPID=Org2MSP
ORG2_CONNECTION_PROFILE=connection/connection-org2.local.yaml
ORG1_WALLET=assets/walletOrg1
ORG2_WALLET=assets/walletOrg2
ORG2_CA_NAME=rca-org1
ORG2_CA_ENROLLMENT_ID_ADMIN=rca-org2-admin
ORG2_CA_ENROLLMENT_SECRET_ADMIN=rca-org2-adminPW



