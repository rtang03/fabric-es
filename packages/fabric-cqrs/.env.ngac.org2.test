## unsure if this is required
COLLECTION=Org1PrivateDetails
PATH_TO_CHANNEL_CONFIG=../../artifacts/channel-config

### MANDATORY
#Orgnaization Admin EnrollmentID
ORG_ADMIN_ID=Admin@org2.example.com
#Organization Admin EnrollSecret
ORG_ADMIN_SECRET=peer1pw
#URL for Orgnization CA
ORG_CA_URL=https://0.0.0.0:5055
# org name defined in connnection profile; case sensitive
ORGNAME=Org2
# Path to Fabric Network
NETWORK_LOCATION=../../network/hosts
# Path to Connection Path
CONNECTION_PROFILE=connection/peer0org2.yaml
# Relative path to wallet directory
WALLET=assets/wallet
# Channel Event Hub
CHANNEL_HUB=peer0.org2.example.com
# Optional:
CHAINCODE_ID=eventstore
CHAINCODE_ID_PRIVATEDATA=privatedata
# Used in logger
APPLICATION=peer0org2app


