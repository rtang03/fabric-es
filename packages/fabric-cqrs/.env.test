#MSPID=Org1MSP
#CHANNEL_NAME=mychannel

COLLECTION=Org1PrivateDetails
PATH_TO_CHANNEL_CONFIG=../../artifacts/channel-config
CHANNEL_NAME=eventchannel

### MANDATORY
#Orgnaization Admin EnrollmentID
ORG_ADMIN_ID=Admin@org1.example.com
#Organization Admin EnrollSecret
ORG_ADMIN_SECRET=peer1pw
#URL for Orgnization CA
ORG_CA_URL=https://0.0.0.0:5054
# org name defined in connnection profile; case sensitive
ORGNAME=Org1
# Path to Fabric Network
NETWORK_LOCATION=../../network/hosts
# Path to Connection Path
CONNECTION_PROFILE=connection/peer0org1.yaml
# Relative path to wallet directory
WALLET=assets/wallet
# Channel Event Hub
CHANNEL_HUB=peer0.org1.example.com
# Optional:
CHAINCODE_ID=eventstore
CHAINCODE_ID_PRIVATEDATA=privatedata


