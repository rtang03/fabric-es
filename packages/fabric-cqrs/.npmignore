*
!dist/
!dist/**/*
!package.json
!*.md
!LICENSE
!README.md
