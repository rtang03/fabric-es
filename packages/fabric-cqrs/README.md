# fabric-cqrs

## Installation

```shell script
yarn install @fabric-es/fabric-cqrs
```
