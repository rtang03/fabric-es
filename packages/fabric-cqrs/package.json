{
  "name": "@espresso/fabric-cqrs",
  "version": "1.0.0",
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "scripts": {
    "build": "yarn run clean && tsc",
    "build:prod": "yarn run clean && tsc -p ./tsconfig.prod.json",
    "package": "yarn run build:cleanup && yarn run cp:package.json && yarn run cp:dist",
    "build:cleanup": "rimraf ../../.build/packages/fabric-cqrs && mkdir -p ../../.build/packages/fabric-cqrs",
    "cp:package.json": "cp ./package.json ../../.build/packages/fabric-cqrs/",
    "cp:dist": "cp -R ./dist ../../.build/packages/fabric-cqrs/",
    "clean": "rimraf dist",
    "test:account": "jest account.unit.test",
    "test:service.integration": "jest service.integration.test",
    "test:service.multiusers": "rimraf assets/walletOrg1 && rimraf assets/walletOrg2 && jest service.multiusers.test",
    "test:service.privatedata.integration": "jest service.privatedata.integration.test",
    "test:service.ngac.crud.integration": "(broke) jest service.ngac.crud.integration.test",
    "test:service.ngac.pe.int.1u": "(broke) jest one-user.service.ngac.pe.int.test",
    "test:store.integration": "jest command.integration.test",
    "test:store.privatedata.integration": "jest command.privatedata.integration.test",
    "test:peer.integration": "jest peer.integration.test",
    "test:peer.privatedata.integration": "jest peer.privatedata.integration.test",
    "test:peer.ngac.integration": "(not implemented yet) jest peer.ngac.integration.test",
    "test:all-unit-test": "jest unit.test",
    "enrollAdmin": "ts-node src/account/enrollAdmin.ts",
    "enrollCaAdmin": "ts-node src/account/enrollCaAdmin.ts",
    "enrollAdminOrg2": "(broke) ts-node src/ngac_test/enrollAdmin.ts",
    "typedoc": "typedoc --excludeNotExported --theme markdown --exclude \"**/*+(.test|constants).ts\" --out docs src/index.ts"
  },
  "engines": {
    "node": ">=8",
    "npm": ">=5"
  },
  "private": true,
  "engineStrict": true,
  "dependencies": {
    "@espresso/operator": "1.0.0",
    "dotenv": "^8.1.0",
    "fabric-client": "^1.4.4",
    "fabric-contract-api": "^1.4.2",
    "fabric-network": "^1.4.4",
    "js-yaml": "^3.13.1",
    "lodash": "^4.17.15",
    "prettyjson": "^1.2.1",
    "redux": "^4.0.4",
    "redux-observable": "^1.2.0",
    "rxjs": "^6.5.3",
    "winston": "^3.2.1"
  },
  "devDependencies": {
    "@types/bytebuffer": "^5.0.40",
    "@types/dotenv": "^6.1.1",
    "@types/jest": "^24.0.18",
    "@types/js-yaml": "^3.12.1",
    "@types/lodash": "^4.14.138",
    "@types/node": "^8.10.54",
    "jest": "^24.9.0",
    "prettier": "^1.19.1",
    "rimraf": "^3.0.0",
    "ts-jest": "^24.0.2",
    "ts-node": "^8.3.0",
    "tslint": "^5.20.0",
    "tslint-config-prettier": "^1.18.0",
    "tslint-plugin-prettier": "^2.0.1",
    "typedoc": "^0.15.7",
    "typedoc-plugin-markdown": "^2.2.16",
    "typescript": "^3.7.2"
  }
}
