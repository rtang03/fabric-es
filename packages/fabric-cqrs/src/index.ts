/**
 * @packageDocumentation
 * @hidden
 */
export {
  createPeer,
  getMockRepository,
  getPrivatedataMockRepository,
  createProjectionDb,
  createQueryDatabase
} from './peer';
export { getContract, getNetwork, submit, evaluate, evaluate$ } from './services';
export * from './types';
export {
  dispatchResult,
  generateToken,
  getAction,
  getErrorAction,
  getErrorActionHandler,
  getSuccessAction,
  getSuccessActionHandler
} from './cqrs/utils';
export { fromCommitsToGroupByEntityId, getHistory, isCommit } from './peer/utils';
