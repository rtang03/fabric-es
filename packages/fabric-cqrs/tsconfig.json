{
  "compilerOptions": {
    "outDir": "dist",
    "target": "es2017",
    "moduleResolution": "node",
    "esModuleInterop": true,
    "module": "commonjs",
    "declaration": true,
    "sourceMap": false,
    "types": ["node", "jest"],
    "lib": ["es2017", "dom", "esnext.asynciterable"],
    "rootDir": "src",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true
  },
  "exclude": ["node_modules", "dist"],
  "typedocOptions": {
    "entryPoints": ["src/index.ts"],
    "out": "docs",
    "name": "fabric-cqrs",
    "includeVersion": true,
    "excludeExternals": true,
    "exclude": "\"**/*+(.test|constants|index).ts\"",
    "readme": "../../README.md",
    "hideGenerator": true
  }
}
