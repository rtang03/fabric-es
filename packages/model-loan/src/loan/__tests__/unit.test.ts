import { buildFederatedSchema } from '@apollo/federation';
import { Commit, getMockRepository, getReducer } from '@fabric-es/fabric-cqrs';
import { DataSrc } from '@fabric-es/gateway-lib';
import { ApolloServer } from 'apollo-server';
import { createTestClient } from 'apollo-server-testing';
import gql from 'graphql-tag';
import {
  APPROVE_LOAN, APPLY_LOAN, CANCEL_LOAN, EXPIRE_LOAN,
  Loan, LoanEvents, loanReducer, loanResolvers, loanTypeDefs,
  REJECT_LOAN, RETURN_LOAN, UPDATE_LOAN
} from '..';

const GET_BY_ID = gql`
query GetLoanById($loanId: String!) {
  getLoanById(loanId: $loanId) {
    loanId
    ownerId
    description
    reference
    comment
    status
  }
}`;
const GET_LOANS_BY_PAGE = gql`
  query GetLoansByPage($pageSize: Int) {
    getPaginatedLoans(pageSize: $pageSize) {
      total
      hasMore
      entities {
        loanId
        ownerId
        description
        reference
        comment
        status
      }
    }
  }
`;

const userId = 'unitTestUser';
const mockdb: Record<string, Commit> = {};
const loanRepo = getMockRepository<Loan, LoanEvents>(mockdb, 'loan', getReducer<Loan, LoanEvents>(loanReducer));

let service;

beforeAll(async () => {
  service = new ApolloServer({
    schema: buildFederatedSchema([{ typeDefs: loanTypeDefs, resolvers: loanResolvers }]),
    dataSources: () => ({
      loan: new DataSrc({ repo: loanRepo })
    }),
    context: () => ({ enrollmentId: 'admin' })
  });
});

afterAll(async () =>
  new Promise(done =>
    setTimeout(() => {
      console.log('Loan Unit Test - Resolver Finished');
      done();
    }, 500)
));

describe('Loan Unit Test - Resolver', () => {
  it('apply loan 0', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L0000',
        description: 'Unit test loan 0',
        reference: 'REF-UNIT-TEST-LOAN-0',
        comment: 'Hello 0000'
      }})
    .then(({ data }) => expect(data.applyLoan.id).toEqual('L0000'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('apply loan 1', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L0001',
        description: 'Unit test loan 1',
        reference: 'REF-UNIT-TEST-LOAN-1',
        comment: 'Hello 0001'
      }})
    .then(({ data }) => expect(data.applyLoan.id).toEqual('L0001'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('apply loan 2', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L0002',
        description: 'Unit test loan 2',
        reference: 'REF-UNIT-TEST-LOAN-2'
      }})
    .then(({ data }) => expect(data.applyLoan.id).toEqual('L0002'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('apply loan 3', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L0003',
        description: 'Unit test loan 3',
        reference: 'REF-UNIT-TEST-LOAN-3',
        comment: 'Hello 0003'
      }})
    .then(({ data }) => expect(data.applyLoan.id).toEqual('L0003'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('apply loan 4', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L0004',
        description: 'Unit test loan 4',
        reference: 'REF-UNIT-TEST-LOAN-4',
        comment: 'Hello 0004'
      }})
    .then(({ data }) => expect(data.applyLoan.id).toEqual('L0004'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('apply loan 5', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L0005',
        description: 'Unit test loan 5',
        reference: 'REF-UNIT-TEST-LOAN-5',
        comment: 'Hello 0005'
      }})
    .then(({ data }) => expect(data.applyLoan.id).toEqual('L0005'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('apply loan 6', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L0006',
        description: 'Unit test loan 6',
        reference: 'REF-UNIT-TEST-LOAN-6',
        comment: 'Hello 0006'
      }})
    .then(({ data }) => expect(data.applyLoan.id).toEqual('L0006'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('apply loan 7', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L0007',
        description: 'Unit test loan 7',
        reference: 'REF-UNIT-TEST-LOAN-7',
        comment: 'Hello 0007'
      }})
    .then(({ data }) => expect(data.applyLoan.id).toEqual('L0007'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('apply loan 8', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L0008',
        description: 'Unit test loan 8',
        reference: 'REF-UNIT-TEST-LOAN-8',
        comment: 'Hello 0008'
      }})
    .then(({ data }) => expect(data.applyLoan.id).toEqual('L0008'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('apply loan 9', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L0009',
        description: 'Unit test loan 9',
        reference: 'REF-UNIT-TEST-LOAN-9',
        comment: 'Hello 0009'
      }})
    .then(({ data }) => expect(data.applyLoan.id).toEqual('L0009'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('apply loan 10', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L0010',
        description: 'Unit test loan 10',
        reference: 'REF-UNIT-TEST-LOAN-A',
        comment: 'Hello 0010'
      }})
    .then(({ data }) => expect(data.applyLoan.id).toEqual('L0010'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('apply loan 11', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L0011',
        description: 'Unit test loan 11',
        reference: 'REF-UNIT-TEST-LOAN-B',
        comment: 'Hello 0011'
      }})
    .then(({ data }) => expect(data.applyLoan.id).toEqual('L0011'))
    .catch(_ => expect(false).toBeTruthy())
  );

  // TODO: Implement lifecycle event attribute to prevent creating same entity more than once
  // NOTE: This 'apply loan' call should return normal, but querying 'L0000' should return the original result instead of the changed values
  it('apply loan 0 again', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L0000',
        description: 'Unit test loan 0VERWRITTEN',
        reference: 'REF-UNIT-TEST-LOAN-0VERWRITTEN',
        comment: 'Hello 0000VERWRITTEN'
      }})
    .then(({ data }) => expect(data.applyLoan.id).toEqual('L0000'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('update description of loan 2', async () =>
    createTestClient(service).mutate({
      mutation: UPDATE_LOAN,
      variables: {
        userId, loanId: 'L0002',
        description: 'Unit test loan 2 EDITED'
      }})
    .then(({ data }) => expect(data.updateLoan.map(d => (d && d.id) ? d.id : '')).toContain('L0002'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('update comment of loan 3', async () =>
    createTestClient(service).mutate({
      mutation: UPDATE_LOAN,
      variables: {
        userId, loanId: 'L0003',
        comment: 'Hello 0003 EDITED'
      }})
    .then(({ data }) => expect(data.updateLoan.map(d => (d && d.id) ? d.id : '')).toContain('L0003'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('update description and comment of loan 4', async () =>
    createTestClient(service).mutate({
      mutation: UPDATE_LOAN,
      variables: {
        userId, loanId: 'L0004',
        description: 'Unit test loan 4 EDITED',
        comment: 'Hello 0004 EDITED'
      }})
    .then(({ data }) => expect(data.updateLoan.map(d => (d && d.id) ? d.id : '')).toContain('L0004'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('Add comment to loan 2', async () =>
    createTestClient(service).mutate({
      mutation: UPDATE_LOAN,
      variables: {
        userId, loanId: 'L0002',
        comment: 'Hello 0002 ADDED'
      }})
    .then(({ data }) => expect(data.updateLoan.map(d => (d && d.id) ? d.id : '')).toContain('L0002'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('create a loan without description', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L9999',
        reference: 'REF-UNIT-TEST-LOAN-9'
      }})
    .then(({ errors }) => expect(errors.reduce((acc, cur) =>
      cur.message.includes('was not provided') ? cur.message : acc, '')).toContain('was not provided'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('create a loan with empty description', async () =>
    createTestClient(service).mutate({
      mutation: APPLY_LOAN,
      variables: {
        userId, loanId: 'L9999',
        reference: 'REF-UNIT-TEST-LOAN-9',
        description: ''
      }})
    .then(({ errors }) => expect(errors.reduce((acc, cur) =>
      cur.message.includes('REQUIRED_DATA_MISSING') ? cur.message : acc, '')).toContain('REQUIRED_DATA_MISSING'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('update an non-existing loan', async () =>
    createTestClient(service).mutate({
      mutation: UPDATE_LOAN,
      variables: {
        userId, loanId: 'L9999',
        comment: 'Hello 9999'
      }})
    .then(({ errors }) => expect(errors.reduce((acc, cur) =>
      cur.message.includes('LOAN_NOT_FOUND') ? cur.message : acc, '')).toContain('LOAN_NOT_FOUND'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('update loan 5 with empty description', async () =>
    createTestClient(service).mutate({
      mutation: UPDATE_LOAN,
      variables: {
        userId, loanId: 'L0005',
        description: ''
      }})
    .then(({ errors }) => expect(errors.reduce((acc, cur) =>
      cur.message.includes('REQUIRED_DATA_MISSING') ? cur.message : acc, '')).toContain('REQUIRED_DATA_MISSING'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('update loan 6 with both successful and fail cases', async () =>
    createTestClient(service).mutate({
      mutation: UPDATE_LOAN,
      variables: {
        userId, loanId: 'L0006',
        reference: 'HI',
        description: '',
        comment: 'Hello 0006 EDITED'
      }})
    .then(({ data, errors }) => {
      const errs = errors.map(e => e.message);
      expect(errs).toContain('Error: INVALID_OPERATION');
      expect(errs).toContain('Error: REQUIRED_DATA_MISSING');
      expect(data.updateLoan.map(d => (d && d.id) ? d.id : '')).toContain('L0006');
    })
    .catch(_ => expect(false).toBeTruthy())
  );

  it('cancel loan 7', async () =>
    createTestClient(service).mutate({
      mutation: CANCEL_LOAN,
      variables: { userId, loanId: 'L0007' }
    })
    .then(({ data }) => expect(data.cancelLoan.id).toEqual('L0007'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('approve loan 8', async () =>
    createTestClient(service).mutate({
      mutation: APPROVE_LOAN,
      variables: { userId, loanId: 'L0008' }
    })
    .then(({ data }) => expect(data.approveLoan.id).toEqual('L0008'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('return loan 9', async () =>
    createTestClient(service).mutate({
      mutation: RETURN_LOAN,
      variables: { userId, loanId: 'L0009' }
    })
    .then(({ data }) => expect(data.returnLoan.id).toEqual('L0009'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('reject loan 10', async () =>
    createTestClient(service).mutate({
      mutation: REJECT_LOAN,
      variables: { userId, loanId: 'L0010' }})
    .then(({ data }) => expect(data.rejectLoan.id).toEqual('L0010'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('flag loan 11 as expired', async () =>
    createTestClient(service).mutate({
      mutation: EXPIRE_LOAN,
      variables: { userId, loanId: 'L0011' }})
    .then(({ data }) => expect(data.expireLoan.id).toEqual('L0011'))
    .catch(_ => expect(false).toBeTruthy())
  );

  it('query loan 0 by id', async () =>
    createTestClient(service).query({
      query: GET_BY_ID,
      variables: { loanId: 'L0000' }
    })
    .then(({ data }) => expect(data.getLoanById).toMatchSnapshot())
    .catch(_ => expect(false).toBeTruthy())
  );

  it('query loan 1 by id', async () =>
    createTestClient(service).query({
      query: GET_BY_ID,
      variables: { loanId: 'L0001' }
    })
    .then(({ data }) => expect(data.getLoanById).toMatchSnapshot())
    .catch(_ => expect(false).toBeTruthy())
  );

  it('query loan 2 by id', async () =>
    createTestClient(service).query({
      query: GET_BY_ID,
      variables: { loanId: 'L0002' }
    })
    .then(({ data }) => expect(data.getLoanById).toMatchSnapshot())
    .catch(_ => expect(false).toBeTruthy())
  );

  it('query loan 3 by id', async () =>
    createTestClient(service).query({
      query: GET_BY_ID,
      variables: { loanId: 'L0003' }
    })
    .then(({ data }) => expect(data.getLoanById).toMatchSnapshot())
    .catch(_ => expect(false).toBeTruthy())
  );

  it('query loan 4 by id', async () =>
    createTestClient(service).query({
      query: GET_BY_ID,
      variables: { loanId: 'L0004' }
    })
    .then(({ data }) => expect(data.getLoanById).toMatchSnapshot())
    .catch(_ => expect(false).toBeTruthy())
  );

  it('query loans by page', async () =>
    createTestClient(service).query({
      query: GET_LOANS_BY_PAGE
    })
    .then(({ data }) => expect(data.getPaginatedLoans).toMatchSnapshot())
    .catch(_ => expect(false).toBeTruthy())
  );
});
