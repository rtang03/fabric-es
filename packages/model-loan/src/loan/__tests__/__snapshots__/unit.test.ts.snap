// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loan Unit Test - Resolver query loan 0 by id 1`] = `
Object {
  "comment": "Hello 0000VERWRITTEN",
  "description": "Unit test loan 0VERWRITTEN",
  "loanId": "L0000",
  "ownerId": "unitTestUser",
  "reference": "REF-UNIT-TEST-LOAN-0VERWRITTEN",
  "status": 0,
}
`;

exports[`Loan Unit Test - Resolver query loan 1 by id 1`] = `
Object {
  "comment": "Hello 0001",
  "description": "Unit test loan 1",
  "loanId": "L0001",
  "ownerId": "unitTestUser",
  "reference": "REF-UNIT-TEST-LOAN-1",
  "status": 0,
}
`;

exports[`Loan Unit Test - Resolver query loan 2 by id 1`] = `
Object {
  "comment": "Hello 0002 ADDED",
  "description": "Unit test loan 2 EDITED",
  "loanId": "L0002",
  "ownerId": "unitTestUser",
  "reference": "REF-UNIT-TEST-LOAN-2",
  "status": 0,
}
`;

exports[`Loan Unit Test - Resolver query loan 3 by id 1`] = `
Object {
  "comment": "Hello 0003 EDITED",
  "description": "Unit test loan 3",
  "loanId": "L0003",
  "ownerId": "unitTestUser",
  "reference": "REF-UNIT-TEST-LOAN-3",
  "status": 0,
}
`;

exports[`Loan Unit Test - Resolver query loan 4 by id 1`] = `
Object {
  "comment": "Hello 0004 EDITED",
  "description": "Unit test loan 4 EDITED",
  "loanId": "L0004",
  "ownerId": "unitTestUser",
  "reference": "REF-UNIT-TEST-LOAN-4",
  "status": 0,
}
`;

exports[`Loan Unit Test - Resolver query loans by page 1`] = `
Object {
  "hasMore": true,
  "items": Array [
    Object {
      "comment": "Hello 0000VERWRITTEN",
      "description": "Unit test loan 0VERWRITTEN",
      "loanId": "L0000",
      "ownerId": "unitTestUser",
      "reference": "REF-UNIT-TEST-LOAN-0VERWRITTEN",
      "status": 0,
    },
    Object {
      "comment": "Hello 0001",
      "description": "Unit test loan 1",
      "loanId": "L0001",
      "ownerId": "unitTestUser",
      "reference": "REF-UNIT-TEST-LOAN-1",
      "status": 0,
    },
    Object {
      "comment": "Hello 0002 ADDED",
      "description": "Unit test loan 2 EDITED",
      "loanId": "L0002",
      "ownerId": "unitTestUser",
      "reference": "REF-UNIT-TEST-LOAN-2",
      "status": 0,
    },
    Object {
      "comment": "Hello 0003 EDITED",
      "description": "Unit test loan 3",
      "loanId": "L0003",
      "ownerId": "unitTestUser",
      "reference": "REF-UNIT-TEST-LOAN-3",
      "status": 0,
    },
    Object {
      "comment": "Hello 0004 EDITED",
      "description": "Unit test loan 4 EDITED",
      "loanId": "L0004",
      "ownerId": "unitTestUser",
      "reference": "REF-UNIT-TEST-LOAN-4",
      "status": 0,
    },
    Object {
      "comment": "Hello 0005",
      "description": "Unit test loan 5",
      "loanId": "L0005",
      "ownerId": "unitTestUser",
      "reference": "REF-UNIT-TEST-LOAN-5",
      "status": 0,
    },
    Object {
      "comment": "Hello 0006 EDITED",
      "description": "Unit test loan 6",
      "loanId": "L0006",
      "ownerId": "unitTestUser",
      "reference": "REF-UNIT-TEST-LOAN-6",
      "status": 0,
    },
    Object {
      "comment": "Hello 0007",
      "description": "Unit test loan 7",
      "loanId": "L0007",
      "ownerId": "unitTestUser",
      "reference": "REF-UNIT-TEST-LOAN-7",
      "status": 1,
    },
    Object {
      "comment": "Hello 0008",
      "description": "Unit test loan 8",
      "loanId": "L0008",
      "ownerId": "unitTestUser",
      "reference": "REF-UNIT-TEST-LOAN-8",
      "status": 2,
    },
    Object {
      "comment": "Hello 0009",
      "description": "Unit test loan 9",
      "loanId": "L0009",
      "ownerId": "unitTestUser",
      "reference": "REF-UNIT-TEST-LOAN-9",
      "status": 3,
    },
    Object {
      "comment": "Hello 0010",
      "description": "Unit test loan 10",
      "loanId": "L0010",
      "ownerId": "unitTestUser",
      "reference": "REF-UNIT-TEST-LOAN-A",
      "status": 4,
    },
    Object {
      "comment": "Hello 0011",
      "description": "Unit test loan 11",
      "loanId": "L0011",
      "ownerId": "unitTestUser",
      "reference": "REF-UNIT-TEST-LOAN-B",
      "status": 5,
    },
  ],
  "total": 12,
}
`;
